clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -701,7 +713,9 @@ private long partitionReady(Cluster cluster, long nowMs, String topic,
                 }
 
                 waitedTimeMs = batch.waitedTimeMs(nowMs);
-                backingOff = batch.attempts() > 0 && waitedTimeMs < retryBackoffMs;
+                backingOff = !batch.hasLeaderChanged(leader) &&
+                        batch.attempts() > 0 &&
+                        waitedTimeMs < retryBackoff.backoff(batch.attempts() - 1);
```

Comment:
This mostly duplicates the code in line 716. Could we pull that into a util method and reuse?

Reply:
Done